### PR TITLE
Examples for Path.Join

### PR DESCRIPTION
**Pull request by rpetrusha**
## Examples for Path.Join

Related to dotnet/docs#7379

### CODE REVIEW

**Comment by BillWagner** [comment]
I tried a couple experiments, and I found some interesting platform considerations for `Path.Combine`:

1. The separator char added by `Path.Combine` is platform dependent:  `\` on Windows, `/` on Linux and Mac.
1. The separator chars in either component are not modified to match the current platform.
1. A separator char from the other platform is not recognized as the separator char when combining paths.  For example:  `Path.Combine(@"C:\docs\", "core");` on Linux produces the string `C:\docs\/core`

We should probably add clarifications on these in the output on the samples.

**Comment by richlander** [comment]
My feedback:

- Worth showing output on both Windows and Linux, unless this is a Windows-only samples.
- Would be good to clarify the difference between join and combine in a comment.
- Most of the scenarios seem to me like edge cases, not best practices. Would be good to make those differences clear.

**Comment by mairaw** [comment]
@rpetrusha what's the status on this one?

**Comment by rpetrusha** [comment]
I working on updating it now, @mairaw.

**Comment by rpetrusha** [comment]
@JeremyKuhne, @mairaw, and @BillWagner, this is finally ready for review.

File: snippets/csharp/api/system.io/path/combine3/Program.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        ShowPathInformation(path1, path2, path3);
+    }
+
+   private static void ShowPathInformation(string path1, string path2, string path3)
+    {
+        var result = Path.Join(path1.AsSpan(), path2.AsSpan(), path3.AsSpan());
+        Console.WriteLine($"Concatenating  '{path1}, '{path2}', and `{path3}'");
+        Console.WriteLine($"   Path.Join:     '{result}'");
+        Console.WriteLine($"   Path.Combine:  '{Path.Combine(path1, path2, path3)}'");
+Console.WriteLine($"   {Path.GetFullPath(result)}");
```
Comment (BillWagner) [inline snippets/csharp/api/system.io/path/combine3/Program.cs]:
```suggestion
        Console.WriteLine($"   {Path.GetFullPath(result)}");
```

**Pull request merged**